pkgs/by-name/dr/draupnir/package.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  pkg-config,
+  sqlite,
+  fetchpatch,
+  matrix-sdk-crypto-nodejs,
+}:
+
+buildNpmPackage rec {
+  pname = "draupnir";
+  version = "2.0.0-beta.6";
+
+  src = fetchFromGitHub {
+    owner = "the-draupnir-project";
+    repo = "Draupnir";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-s1LWXVwY+7LD7cJtZW7mBLsdpB499zS/nDsJ7qaQDfg=";
+  };
+
+  patches = [ ./remove-yarn.patch ];
+
+  postPatch = ''
+    rm yarn.lock
+    cp ${./package-lock.json} ./package-lock.json
+
+    # so that npmInstallHook picks it up
+    substituteInPlace package.json --replace-fail '"main": "' '"bin": "'
+  '';
+
+  npmDepsHash = "sha256-zKfLnxXo383Fjb+xJWoRyFaMrpftYaFhyqPBe6pQ5Ek=";
+
+  npmFlags = [ "--legacy-peer-deps" ];
+  npmRebuildFlags = [ "--build-from-source" ];
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ sqlite ];
+
+  # TODO: move to preNpmRebuild
+  preBuild = ''
+    rm -r node_modules/better-sqlite3/{build,deps/common.gypi,deps/sqlite3.gyp,deps/sqlite3}
```

Comment:
```nix
# Remove things we know we don't need that just increase closure size.
```

Reply:
Isn't that still needed for sqlite?

Reply:
I believe the removed files are all only used by the `better-sqlite3` package to build it's own vendored sqlite. `better-sqlite3-dynamic-link.patch` replaces that with just linking against the nixpkgs sqlite.

Reply:
Correct.